### ISSUE TITLE
Move jax.nn.log_softmax to jax.scipy.special?

### ISSUE THREAD

**Issue by NeilGirdhar** [google/jax#20700]
It's in SciPy now, so it seems like the logical place for it?

**Comment by jakevdp** [comment]
Yes, let's do this (both for `softmax` and `log_softmax`) though we should continue exposing them in `jax.nn` as well.

Would you like to prepare a PR?

**Comment by jakevdp** [comment]
(note the in-flight change to these functions in #20690; might be cleanest to wait until that is merged before moving any sources

**Comment by NeilGirdhar** [comment]
Sure, I'm happy to do that.
